### PR TITLE
[cmake] try to improve the CMake find package mechanism for the preco…

### PR DESCRIPTION
**Pull request by AchimTuran**
…mpiled Windows binaries

## Description

I had some issues on my Windows nodes to compile Kodi because CMake didn't find the correct precompiled binaries like Python or iconv.

This solutions adds some hints if the CMake variable `DEPENDS_DIR` is set during the CMake configure. e.g. a CMake call could look like `cmake -G "Visual Studio 15 Win64" -T host=x64 ../xbmc  -DDEPENDS_DIR=<repo>/project/BuildDependencies/x64`.

## Motivation and Context

I know it is not a clean solution but one that works and avoids to mess around with environment variables. But one that worked on my machines and is an intermidiate solution until the Windows dependency rework is done.

## Types of change

- [X] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by AchimTuran** [comment]
@AlwinEsch 
Can you please test this PR and give feedback if it also works on your machines.

**Review by Rechi — CHANGES_REQUESTED** [review]
This is not correct, because now you pull information about the dependency layout into the general CMake find modules.

There are also some behaviour changes for non Windows systems, which could break building Kodi.

**Comment by AchimTuran** [comment]
@Rechi 
As I said it is not a clean solution.

File: cmake/modules/FindPython.cmake [hunk 1]
```diff
@@ -10,8 +10,8 @@ if(PKG_CONFIG_FOUND)
 endif()
 
 find_program(PYTHON_EXECUTABLE python3 ONLY_CMAKE_FIND_ROOT_PATH)
-find_library(PYTHON_LIBRARY NAMES python3.7 python3.6 python3.5 PATHS ${PC_PYTHON_LIBDIR})
```
Comment (AchimTuran) [inline cmake/modules/FindPython.cmake]: @Rechi 
So far I know these two variables are not used anywehre. Do you know why they are present here?

Comment (Rechi) [reply cmake/modules/FindPython.cmake]: `PC_PYTHON_LIBDIR` and `PC_PYTHON_INCLUDE_DIRS` is set by `pkg_check_modules(PC_PYTHON python3>=3.5 QUIET)` a few lines above.

Comment (AchimTuran) [reply cmake/modules/FindPython.cmake]: Okay didn't know that these variables are set from the PKG call.
What would be a lean solution to force the paths from our precompiled binary folders? On my machine somehow the system libraries are preferred.

**Comment by Rechi** [comment]
If you look at the search order for [find_library](https://cmake.org/cmake/help/v3.5/command/find_library.html) and [find_path](https://cmake.org/cmake/help/v3.5/command/find_path.html), it should already search the dependencies folder first, because of https://github.com/xbmc/xbmc/blob/615a55ea976c627778752d4627dc842be7adf1d5/cmake/scripts/windows/ArchSetup.cmake#L37

Please check the value of `CMAKE_PREFIX_PATH` variable just before the `find_library` and `find_path` calls. Also share `CMakeCache.txt` of a clean configure without any modifications.
For the Python the wrong detection could be caused by `find_package(PythonLibs 3.5 REQUIRED)`.

**Comment by AchimTuran** [comment]
https://github.com/xbmc/xbmc/pull/16905 solved this my issues, which is the reason why I close this PR.

**Pull request closed without merging**